cxx-squid/src/main/java/org/sonar/cxx/preprocessor/CxxPreprocessor.java

```diff
@@ -127,34 +162,56 @@ public CxxPreprocessor(SquidAstVisitorContext<Grammar> context,
 
     pplineParser = CppParser.create(conf);
 
+    final List<String> forcedDefines = conf.getDefines();
+    Map<String, Macro> forcedMacros = Collections.emptyMap();
+    if (!forcedDefines.isEmpty()) {
+      LOG.debug("parsing forced defines");
+      forcedMacros = parseMacroDefinitions(forcedDefines);
+    }
+
+    // fill CxxPreprocessor.fixedMacros
+    // getMacros() returns CxxPreprocessor.fixedMacros
+    if (getMacros() != fixedMacros) {
+      throw new IllegalStateException("expected fixedMacros as active macros map");
+    }
     try {
       getMacros().setHighPrio(true);
-
-      // parse the configured defines and store into the macro library
-      for (String define : conf.getDefines()) {
-        LOG.debug("parsing external macro: '{}'", define);
-        if (!"".equals(define)) {
-          Macro macro = parseMacroDefinition("#define " + define);
-          LOG.debug("storing external macro: '{}'", macro);
-          getMacros().put(macro.name, macro);
-        }
-      }
-
-      // set standard macros
+      getMacros().putAll(forcedMacros);
```

Comment:
Is this the right order? First standard macros and then forced? Think should be possible to overwrite standard macros with forced.